homeassistant/components/homematicip_cloud/event.py

```diff
@@ -42,23 +50,26 @@ async def async_setup_entry(
     """Set up the HomematicIP cover from a config entry."""
     hap = hass.data[DOMAIN][config_entry.unique_id]
 
-    async_add_entities(
-        HomematicipDoorBellEvent(
-            hap,
-            device,
-            channel.index,
-            EVENT_DESCRIPTIONS["doorbell"],
+    for description in EVENT_DESCRIPTIONS.values():
+        async_add_entities(
+            HomematicipDoorBellEvent(
+                hap,
+                device,
+                channel.index,
+                description,
+            )
+            for device in hap.home.devices
+            for channel in device.functionalChannels
+            if description.channel_selector_fn
+            and description.channel_selector_fn(channel)
         )
```

Comment:
Please only call `async_add_entities` once by adding the extra for loop to the list comprehension

Reply:
argh...sorry. my bad.